packages/date-picker/src/DatePicker/DatePickerInput/DatePickerInput.tsx

```diff
@@ -73,17 +73,21 @@ export const DatePickerInput = forwardRef<HTMLDivElement, DatePickerInputProps>(
      * Called when the input, or any of its children, is clicked.
      * Opens the menu and focuses the appropriate segment
      */
-    const handleInputClick: MouseEventHandler<HTMLElement> = e => {
+    const handleInputClick: MouseEventHandler<HTMLInputElement> = e => {
       if (!disabled) {
         openMenu(e);
         const { target } = e;
+        e.preventDefault();
         const segmentToFocus = getSegmentToFocus({
           target,
           formatParts,
           segmentRefs,
         });
 
+        // console.log('🐥 click 🐥');
+
         segmentToFocus?.focus();
+        segmentToFocus?.select();
```

Comment:
Do we need both `.focus` and `.select`? (my assumption is that select automatically sets focus, but maybe not)

Reply:
I think so, a bunch of test fail without `.focus`